### DISCUSSION TITLE
Week 2: ML with Python

### DISCUSSION THREAD

**Discussion by BekahHW** [deepgram/community#82] [General help]
## Getting Started

The goal this week is to get through the next 8-10 videos of [freeCodeCamp's ML with python](https://www.freecodecamp.org/learn/machine-learning-with-python/) and check-in here. 

## Learning Resources
_I'll update these as we move through the week._

## Check-in
1. What have you learned about neural networks? 
2. How do you evaluate a model?
3. What do you know today that you didn't when you started?
4. What are some key terms or ideas from these lessons?
5. What do you need to be clarified or what questions do you have?
6. What are you excited to learn more about?

**Comment by BekahHW** [comment]
1. A way to use machine learning that takes input, passes it to a function and produces an output. 
2. You have to identify what you need and the problem you're trying to solve, to determine the best model for the job. 
3. I learned more about the activation functions. I want to learn more about what these mean bc I have a general idea, but I'm not sure of how you decide what type of activation function you need. I'm exploring [this resource](https://machinelearningmastery.com/a-gentle-introduction-to-sigmoid-function/#:~:text=A%20sigmoid%20unit%20in%20a,the%20weighted%20sum%20of%20inputs.) to learn more about sigmoid functions.
4. _A lot._ I think some of it is doing some practical application. I want to go through a learning lab that explains the jargon as part of the creation process.
5. I'm really looking forward to the sections on NLP.

**Comment by paceaux** [comment]
I ended up watching a lot of these more than once. sooo....

1. What did I learn? A neural network is complex (has many parts) but not complicated (difficult). The idea is to pass bits of data from one layer to another and tweak something about it along the way, until they spit something out the end. 
2. How to evaluate a model... you know? I'm still not sure. His discussions of NN were too abstract for me to really grok how to apply it in practice. 
3. What I know know that I didn't before:
     1. _just_ three layers to a neural network (in, out, hidden)
     2. all connections are with weights
     3. layers can have biases
     4. Activation functions are for fixing data
     5. the NN isn't useful when you run it _once_. It's useful when you run it many times, evaluating that Loss each time and relying on a "Gradient Descent" to tell you where you need to adjust weights and values via "back propogation"
6. What are the key terms? 
     1. Input/output/hidden layer
     2. Activation function (could include TanH, Sigmoid, Rectified linear)
     3. Loss Function (Mean Squared, Mean Absolute, Hinge Loss)
     4. Gradient Descent
     5. Back propogation
7. What do I need clarified? _all of it_? I think I need to see how this plays out in a scenario that doesn't involve numbers. 
8. What do I want to learn more of? How to actually apply this against some language research I want to do. AKA how can I apply this to something _interesting_?

**Reply by BekahHW** [reply]
Your answer to #5 is exactly what I've been exploring. I haven't made it to the next section of the videos yet, but I'm hoping the NLP stuff helps me to better understand.

**Reply by paceaux** [reply]
BTW, if you're interested, I'm happy to share a use-case/ scenario that I'm hoping to explore with this.

**Reply by BekahHW** [reply]
@paceaux I'm definitely interested!

**Comment by paceaux** [comment]
gonna say this:

I'm a little uncomfortable not understanding how the Neural Network he used managed to classify those images. (From the "Creating a Model" module). 

that seemed like too much magic to me.

**Reply by BekahHW** [reply]
Totally agree with this. I'm working to compile some resources for next week to slow things down and demystify this process a little more.
